common/src/main/java/tachyon/util/io/BufferUtils.java

```diff
@@ -231,7 +231,7 @@ public static boolean equalIncreasingByteBuffer(int start, int len, ByteBuffer b
       return false;
     }
     buf.rewind();
-    if (buf.remaining() != len) {
+    if (buf.remaining() < len) {
```

Comment:
Could you take a look at the usages and see if there's any instance where we rely on the behavior? I think doing a strict equals is more meaningful.